SDK/AppCenter/Microsoft.AppCenter.Shared/AppCenter.cs

```diff
@@ -18,7 +18,7 @@ public partial class AppCenter
         const string TargetKeyName = "target";
         const string TargetKeyNameUpper = "Target";
         const string AppSecretKeyName = "appsecret";
-        const string SecretsPattern = @"([^;=]+)=([^;=]+);?";
+        const string SecretsPattern = @"([^;=]+)=([^;=]+)(;\s*)?";
```

Comment:
1. Why second group contains "not `=`"? Do we have the test for `id=123=123`?
2. In case (1), new line will be included into the first group of next match
3. We don't need to capture third group. It can be marked as non-captured (`?:`)

My suggestion is
```suggestion
        const string SecretsPattern = @"([^\s;=]+)=([^;]+)(?:;\s*)?";
```

Reply:
1. 2. will try to improve the `=`.

3. we don't want to allow garbage characters with the suggestion you could pass:

`garbabe android=123;ios=456` and android would correctly parse.
with the one in the PR, android would not be recognized as a platform and trigger the error which according to me is better.

Reply:
```suggestion
        const string SecretsPattern = @"([^;=]+)=([^;]+)(?:;\s*)?";
```

Reply:
3. it's about do not capture third group

Reply:
I was talking about your first code suggestion that I disagreed. The new suggestion is ok.